### ISSUE TITLE
JSON array element removal syntax is not supported

### ISSUE THREAD

**Issue by parris** [oguimbal/pg-mem#77]
Hello! Just want to say this library is awesome again and I found a bug with JSONB array subtractions.

Here's the most simple example I could come up with that is not supported: 

```
        const example = db.public.one(`select (('{"a": ["b", "c"]}'::jsonb) -> 'a')::jsonb - 'b' as test`);
        expect(example.test.length).to.equal(1);
        expect(example.test[0]).to.be('c');
```

The above code will produce the following error:

```
SyntaxError: Unexpected token b in JSON at position 0

💥 This is a nasty error, which was unexpected by pg-mem. Also known "a bug" 😁 Please file an issue !

*️⃣ Failed SQL statement: select (('{"a": ["b", "c"]}'::jsonb) -> 'a')::jsonb - 'b' as test;
```

Here's a bit of the stacktrace:

```
    at JSON.parse (<anonymous>)
    at value.setConversion.toJsonb.toJsonb (node_modules/pg-mem/src/datatypes/t-jsonb.ts:52:48)
    at Evaluator.val (node_modules/pg-mem/src/evaluator.ts:133:28)
    at new Evaluator (node_modules/pg-mem/src/evaluator.ts:97:33)
    at Evaluator.setConversion (node_modules/pg-mem/src/evaluator.ts:121:16)
    at JSONBType.doBuildFrom (node_modules/pg-mem/src/datatypes/t-jsonb.ts:52:22)
    at TextType.convert (node_modules/pg-mem/src/datatypes/datatype-base.ts:170:28)
    at Evaluator.convert (node_modules/pg-mem/src/evaluator.ts:218:26)
    at buildBinary (node_modules/pg-mem/src/expression-builder.ts:205:29)
    at _buildValueReal (node_modules/pg-mem/src/expression-builder.ts:70:20)
```

It may also be helpful to know the full use case. I'm effectively trying to remove 1 array element by value during an update. Here's a unit test I put together:

```
  it('supports jsonb array subtraction by value', () => {
        const db = newDb();
          pgMemHelpers(db);
          db.public.none(`
            create table page(
              id int,
              doc jsonb
            );
            insert into page values (1, '{ "hello": { "world": ["a", "b", "c"]} }');
          `);
          const example = db.public.one(`select (('{"a": ["b", "c"]}'::jsonb) -> 'a')::jsonb - 'b' as test`);
          expect(example.test.length).to.equal(1);
          expect(example.test[0]).to.be('c');

        const page1 = db.public.one(`
          select (doc->'hello'->'world' - 'a') as subset from public.page WHERE id = 1;
        `);
        expect(page1.subset.length).to.equal(2);

        db.public.one(`
          UPDATE public.page SET doc = jsonb_set(doc, '{hello, world}', doc->'hello'->'world' - 'a', true) WHERE id = 1;
        `);
        const page2 = db.public.one(`select * from page where id = 1;`);
        expect(page2.doc.hello.world.length).to.equal(2);
      });
 ```

'pgMemHelpers' above is the same functions here - https://github.com/oguimbal/pg-mem/discussions/74

**Comment by oguimbal** [comment]
Thanks for the report ! I will fix it asap

**Comment by RomThi** [comment]
Hello !
I have the same issue.
Do you have any news ?
Thanks a lot !

**Comment by oguimbal** [comment]
Hi, no, I had no time to work on pg-mem lately.
I'll dedicate some time to it as soon as my schedule clears a bit, but I dont know when, unfortunately :(

Note to myself, or whoever wants to attempts implementing this: This issue requires to implement binary operators which support heterogenous input types, like in #119 and #134... meaning that [this](https://github.com/oguimbal/pg-mem/blob/52deb071381b0db12239b55be4d6ad00580b3d74/src/expression-builder.ts#L208-L366) should be refactored in something generic & extensible.

**Comment by oguimbal** [comment]
Hi, this should be fixed in `pg-mem@2.2.0` (reopen if not !)

Sorry for the delay !
